### DISCUSSION TITLE
Adding Printer Profile

### DISCUSSION THREAD

**Discussion by choice3d** [OrcaSlicer/OrcaSlicer#1290] [General]
Hi All,

Sorry if this has been asked before, I couldn't find anything related. 

How do I add printer profiles to the slicer from here (github)? I want to add the E5 S1 profile but not sure what to copy/download and where to put it.

I'm using V1.6.2

Cheers

**Comment by fr3ak2402** [comment]
you must place your files in your fork in the folder Resources/profiles....

Then you can make a PR....

The rest will be done by others

**Reply by ZakVilneuv** [reply]
Hi.
I've just searched in the issues, PR, docs, etc... but didn't find any information about that (I hope I didn' miss it).

What kind of files do you need us to put into the "Ressources/profiles" folder ?
I've seen some "cover" and "buildplate_texture" png files, and "buildplate_model" stl files.
I understand why, but don't really know how to build  the stl files for example (thickness ?).
And about the buildplate texture, my Creality CR-10S PRO (upgraded to V2) has a very simple rounded corners nearly square build plate with a Creality logo in the middle. Do you need something special for this PNG ?

Is there a kind of document detailing requirements for the files that you want to receive from us ?

Do you need any other information ? (printer building volume, etc...)

Please let me know.
Thank you for your work.

**Comment by Aquaman7** [comment]
When will the Creality K1 be added? I hate Creality slicer...

**Reply by fr3ak2402** [reply]
I would finish the profiles at the weekend.

**Reply by fr3ak2402** [reply]
I have finished the profile.

**Reply by ywevis74** [reply]
Will the Creality K1 SE be added?

**Comment by Ferddy2500** [comment]
would love a profile for the voxelab aries

**Reply by fr3ak2402** [reply]
I will also finish it this weekend, I put it on my to-do list.

**Reply by Ferddy2500** [reply]
thanks i love this slicer

**Reply by fr3ak2402** [reply]
I have finished the profile.

**Comment by Ferddy2500** [comment]
Thank you so much 👍


>

**Comment by modulateddreams** [comment]
@fr3ak2402 Any chance of getting the Creality CR-10 Smart Pro added also?

**Reply by myomeroz** [reply]
Hello, did you find a profile for the CR-10 Smart Pro?

**Comment by modulateddreams** [comment]
I have not sorry.

**Reply by myomeroz** [reply]
Why don't they think of us? :((

**Comment by WGoat** [comment]
Hi there. 
Where can a find the profiles doe Ender 3 Pro and the CR-6 SE? Been looking but with no success.

**Comment by Cb3d40T** [comment]
I've been looking for a CR10s Pro v2 profile  i tried one of the others cr10s and added some start gcode but not working out like it should. any help would be great

**Reply by unmystakable** [reply]
Im hoping they make one as well....Im trying to migrate from Cura.

**Reply by Gurba667** [reply]
+1 on this.

**Comment by Pumba763** [comment]
Is there a profile for the Cr10 Smart in the works? I'm trying to migrate from SuperSlicer now that it is dead

**Reply by foreachthing** [reply]
There's that: https://github.com/SoftFever/OrcaSlicer/discussions/1636 and then you can publish your profile.
Or, you could install (or use the portable version) and check if it's already in there...

**Reply by Pumba763** [reply]
sort of works but would be good for someone to do a more complete version.

**Reply by foreachthing** [reply]
I just went through each setting and applied it manually... that has worked pretty well.

**Reply by Pumba763** [reply]
found this for the cr-6se
https://github.com/bhast2/CR-6Se-Max-OrcaSlicer-profile#cr-6se-max-orcaslicer-profile

**Comment by Glenstoen** [comment]
Is there a profile for the Creality CR-10s pro V2 coming?

**Comment by nausfaratu** [comment]
IS it possible to have a profile for Prusa MK3S+ 0.6mm nozzle?

**Comment by BrettPopp** [comment]
Can I please request the Anycubic Kobra 2 Max profile. Would love to use this slicer with that printer. Thank you.

**Comment by livinginkaos** [comment]
Comgrow T500 all 3 nozzles selected and confirmed but the printer profile is not showing up in the printer selection.  Am I missing something?

**Comment by IAmThirdGear** [comment]
If you get time can you get a profile for the Qidi i-fast? I have attached the Prusa slicer profile. Hope this helps.
[PrusaSlicer_fast.zip](https://github.com/SoftFever/OrcaSlicer/files/13691439/PrusaSlicer_fast.zip)

**Comment by fulfordpr** [comment]
I am getting a Kobra 2 Max and would Like to use this as well.

**Reply by Krazymad91** [reply]
please someone add this printer. I have the same one and im still new to all of this

**Comment by BblackmilkK** [comment]
add kobra 2 series ?

**Comment by arcaraway** [comment]
Looking for a profile for a Creality CR-10S. I have a Bambu P1S and want to switch over to Orca from both Creality Slicer and Bambu Studio.

**Comment by galadriann** [comment]
Creality Ender 3 S3 SE
on the start Gcode I found a couple of entries missing that I added my profile and a few changes to where the default start line is setup which helps widen the overall print size:

G420 S1. ;uses saved mesh
M412 S0 ; filament sensor not present - Marlin FW

M413 S0 ;Disable Powr-Loss Recovery

; position default line a little more to the left of the plate.
G1 X-3.5 Y20 Z0.28 F5000.0 ;Move to start position
G1 X-3.5 Y145.0 Z0.28 F1500.0 E15 ;Draw the first line
G1 X-3.2 Y145.0 Z0.28 F5000.0 ;Move to side a little
G1 X-3.2 Y20 Z0.28 F1500.0 E30 ;Draw the second line

hope this helps

**Comment by rajanp-oboe** [comment]
Is there a chance of a profile for Lulzbot Mini 2 standard tool head 0.5 nozzle being added? I've been trying to make my own profile but have been struggling with that. I primarily use lulzbot mini 2 and Ninjatek chinchilla together and the CuraLE edition has profiles for both printer and filament but I see no way to transfer to orcaslicer or input corresponding values.

**Comment by clems77750** [comment]
possible d'ajouter le profil de ma misar s ? aucune geeetech sur orca

**Comment by dan4ik123654** [comment]
would love a profile for the longer lk4 pro. I really liked this slicer

**Comment by muradmalik23** [comment]
Hey, is there any way the creality cr10s pro v2 would be supported on orca?

**Comment by realrelook** [comment]
Good day.!  -issue CR-10S Pro-

I am own the old creality CR-10s Pro (V1) and would like to make a test with OrcaSlicer, but I can not find my printer. Does anyone did setup the CR-10S Pro already and  can share the setup, pleaase.
Thank you.!

**Comment by rodogg88** [comment]
hey, anyone have a profile for a sunlu t3?

**Comment by mccolld1** [comment]
Does anyone know where I can find a filament profile for TPU?  Specifically looking for Sainsmart.  I figure I'll need to tune it but at least I'm not starting from scratch.

**Comment by bsafh** [comment]
I would like to use my old but faithful Geeetech A30 printer (smartto, not Marlin) with Orca, but I cant find any profile for that ... (the A30Pro profile from PrusaSlicer starts printing midair well above the bed...)

Any chance that mit be added?

**Reply by rodogg88** [reply]
Sounds like the printer dimensions need to be corrected in the profile. Is the z size larger than 260 mm? If so, add the correct height

**Reply by bsafh** [reply]
Z size is 420 mm

Corrected, but still it does weird things. The A30Pro is Marlin based, the A30 runs "smartto" (Geeetech special ...), so the gcodes may not match (perhaps no arc?)

**Reply by bsafh** [reply]
OK, I made a bit of a progress. 

It looks like the printer does "speak" Marlin 2, not legacy Mariln and not RepRap. 

I installed the Geeetech Software "EasyPrint3D" - which is horrible, but it get's prints out of the A30.
The config is sparse - but can be exported:
[easyprint3d-a30-settings.ini.txt](https://github.com/user-attachments/files/17179938/easyprint3d-a30-settings.ini.txt)

I tried to construct an Orca profile from that, but ... well, it moves the printhead in the right positions and speeds, but it does not extrude anything - which is not really helpful. Can anyone help me out on that?

[Geeetech A30 0.4 steel nozzle.orca_printer.zip](https://github.com/user-attachments/files/17179952/Geeetech.A30.0.4.steel.nozzle.orca_printer.zip)
(should be a complete preset profile for filament and printer)
[Printer presets.zip](https://github.com/user-attachments/files/17179959/Printer.presets.zip)

My attempts at JSON config  files in Orca are these:

[geeetech_pla.zip](https://github.com/user-attachments/files/17179974/geeetech_pla.zip)
[geeetech_a30.zip](https://github.com/user-attachments/files/17179975/geeetech_a30.zip)

There must be some simple error in my Orca config - movements look fine - but nothing, really nothing is extruded, despite the nozzle temperature being OK and as set in the config.

More or less working start/stop gcode **_in the "EasyPrint3D" Software_** (which does NOT have variable like "`bed_temperature`" ...) looks like this

```
; Start GCODE A30
M117 Preparing....    ; Nozzle clean message
M140 S{material_bed_temperature_layer_0}   ; set bed temp - does not work in EasyPrint3D - no variables ....
M107                  ; Turn layer fan off
M106 ; Max fan to discourage ooze sticking to nozzle
G21                   ; Metric
G28 ; HOME
M420 S1 ; AUTOLEVEL ON
G1 X-5.0 Y40.0 F7200; Links vom Bett positionieren, Position für das Aufheizen einnehmen
G1 Z10.0; Z-Achse senken vor dem Aufheizen
M104 S220 ; preheat extruder temp
M104 S220 ; preheat extruder temp
M140 S60
M109 S220
M117 Heating complete
M117 Begin Autolevel
G29; Auto-Bed-Leveling (auskommentieren oder Loeschen wenn kein Sensor verbaut ist)
;Duese vorbereiten
G92 E0                ; Zero extruder
G28                   ; Home X/Y/Z axis
G1 Z0; Duese senken als Vorbereitung fuer das Material abstreifen
G1 Z0.2; Duese etwas anheben fuer Material Vorlauf
G0 X15 Z0.1 F9000      ; Move up 0.1mm
G0 Y10 F500           ; Move 10mm to get rid of material
G0 Z1 F9000           ; Move up
G0 Y45 F9000         ; Move to line start
G0 Z0.28 F9000         ; Move down
M117 Draw purge lines
G1 Y225 E15 F1500       ; Extrude 15mm filament over 180mm Y axis
G1 X15.8 Y225.0 Z0.28 F5000.0 ; Move to side a little
G1 Y45 Z0.28 F1500.0 E30 ; Draw the second line
G1 Y35 F500           ; Move 10mm without extracting
G92 E0.0 ; reset extruder distance position
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 E-1 F1400          ; Retract 7mm (better 1 or 2) filament
G1 X17 Y45 Z0.3 F5000.0 ; Move over to prevent blob squish
G28
M420 S1
M117 Printing....     ; Printing message
```

```
;END GCODE A30
M117 Ende sequence
G91 ;setze auf relative Positionierung
G1 E-3 F1800 ;Filament zurückziehen
G1 Z10 F9000 ; Z-Achse 10mm hoch
G90 ;setze auf absolute Positionierung
;;G28 X0 ; home X axis
M106 S0 ; turn off cooling fan
M104 S0 ; turn off extruder
M140 S0 ; turn off bed
G1 X-10 Y320 F3000 ; bring bed forward
M84 ; disable motors
M117 Finished!.       ; Printing message
```

**Reply by bsafh** [reply]
Update:

The A30 printer seems to run a variant of **Repetier G-Code**, **not** "Marlin(legacy)". There is no official documentation available from Geeetech unfortunately.

But, at least in Cura the g-code favour is set to "Repetier" and the g-code produced by Cura runs fine on the printer.

The g-code produced by Orca as "Marlin(legacy)" does NOT run, e.g. it uses M73 which the printer does not understand. 

As Orca is based on PrusaSlicer and Bambu Studio, and Prusa Slices does offer Repetier as g-code flavor, why does Orca not offer that?

**Comment by Ferritdiver** [comment]
does anyone know if the Qidi Plus 4 will be added anytime soon?

**Comment by orcatime** [comment]
Hello, I'd like to add a printer profile that isn't in the slicer.

I've created a profile for my Sunlu S8 Plus and I'd like to add it to the printer folder.
My question is, how do I do this?
